source/controllers/ComplicationsController.mc

```diff
@@ -21,7 +22,26 @@ class ComplicationsController {
     );
   }
 
-  function onComplicationUpdate(complicationId as Complications.Id) as Void {
+  hidden function getUnitByEnumValue(index as Complications.Unit?) as String? {
+    if (index == null) {
+      return null;
+    }
+
+    var units =
+      [
+        Rez.Strings.UnknownUnit,
+        Rez.Strings.DistanceUnit,
+        Rez.Strings.ElevationUnit,
+        Rez.Strings.HeightUnit,
+        Rez.Strings.SpeedUnit,
+        Rez.Strings.TemperatureUnit,
+        Rez.Strings.WeightUnit,
+      ] as Array<Symbol>;
```

Comment:
The order of this array comes from the Unit enum definition [here](https://developer.garmin.com/connect-iq/api-docs/Toybox/Complications.html#Unit-module)